### PR TITLE
Make BaseDiskSpec derive from BaseSpec.

### PR DESCRIPTION
**Pull request by skschneider**
Adds option decoder support to BaseDiskSpec.

### CODE REVIEW

**Comment by voellm** [comment]
Can you give an example of how this feature will be used and how this helps?

**Comment by skschneider** [comment]
This change is to address #668.

At the moment, these only decoders used by disk specs are for type-checking, which helps prevent users from passing in bad values via a YAML config file and getting cryptic error messages later in the run when the values are used (e.g. during resource provisioning). The BaseSpec type-checking happens before provisioning begins and can catch mistakes early.

One example of where the decoders are used in BaseVmSpecs is allowing users to provide a a memory value with units ('750MiB', '1.5GiB'). In the future, providers like AWS that allow specifying disk IOPS may benefit from similar flexibility in BaseDiskSpecs.

In the future, the decoders could also be extended to provide a help string describing their accepted values, so this is a step toward auto-documenting available config options.

File: tests/disk_test.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
+# Copyright 2015 PerfKitBenchmarker Authors. All rights reserved.
```
Comment (ehankland) [inline tests/disk_test.py]: nit: Change the dates to 2016 for new files

**Comment by ehankland** [comment]
LGTM

File: tests/disk_test.py (hunk 1 above)
Comment (skschneider) [inline tests/disk_test.py]: Done.

**Pull request merged**